pkg/controller/kubelet-config/kubelet_config_features_test.go

```diff
@@ -29,8 +28,9 @@ func TestFeatureGateDrift(t *testing.T) {
 			if err != nil {
 				t.Errorf("could not generate kubelet config from templates %v", err)
 			}
-			dataURL, _ := dataurl.DecodeString(*kubeletConfig.Contents.Source)
-			originalKubeConfig, _ := decodeKubeletConfig(dataURL.Data)
+			contents, err := ctrlcommon.DecodeIgnitionFileContents(kubeletConfig.Contents.Source, kubeletConfig.Contents.Compression)
+			require.Nil(t, err)
+			originalKubeConfig, _ := decodeKubeletConfig(contents)
```

Comment:
maybe add a require.Nil on the error here too? there wasn't one before but.. seems like there should be. then again feel free to ignore.

Reply:
Yeah, just e.g. 
```
$ rg -g '!vendor/'  ', _ :?='
pkg/operator/status.go
142:            optrVersion, _ = optr.vStore.Get("operator")
217:    optrVersion, _ := optr.vStore.Get("operator")

pkg/server/api_test.go
158:                    headers, _ := parseAcceptHeader(header.input)

pkg/operator/sync.go
227:    optrVersion, _ := optr.vStore.Get("operator")
578:    optrVersion, _ := optr.vStore.Get("operator")
696:                            releaseVersion, _ := optr.vStore.Get("operator")
859:    optrVersion, _ := optr.vStore.Get("operator")

pkg/controller/template/template_controller_test.go
203:            e, _ := expected.(core.CreateAction)
212:            e, _ := expected.(core.UpdateAction)
222:            e, _ := expected.(core.PatchAction)

pkg/controller/container-runtime-config/container_runtime_config_controller_test.go
306:            e, _ := expected.(core.CreateAction)
312:            e, _ := expected.(core.UpdateAction)
321:            e, _ := expected.(core.PatchAction)
370:    blockedRegistries, _ := getValidBlockedRegistries(releaseImageReg, &imgcfg.Spec)
442:                    ctrCfgKey, _ := getManagedKeyCtrCfg(mcp, f.client, ctrcfg1)
478:                    keyCtrCfg, _ := getManagedKeyCtrCfg(mcp, f.client, ctrcfg1)
558:                    keyReg1, _ := getManagedKeyReg(mcp, nil)
559:                    keyReg2, _ := getManagedKeyReg(mcp2, nil)
600:                    keyReg1, _ := getManagedKeyReg(mcp, nil)
601:                    keyReg2, _ := getManagedKeyReg(mcp2, nil)
696:                    keyReg1, _ := getManagedKeyReg(mcp, nil)
697:                    keyReg2, _ := getManagedKeyReg(mcp2, nil)
812:                            keyReg, _ := getManagedKeyReg(pools[i], nil)

pkg/controller/render/render_controller_test.go
146:            e, _ := expected.(core.CreateAction)
155:            e, _ := expected.(core.UpdateAction)
164:            e, _ := expected.(core.PatchAction)

pkg/controller/kubelet-config/kubelet_config_controller_test.go
325:            e, _ := expected.(core.CreateAction)
331:            e, _ := expected.(core.UpdateAction)
340:            e, _ := expected.(core.PatchAction)
394:                    kubeletConfigKey, _ := getManagedKubeletConfigKey(mcp, f.client, kc1)
486:                    kubeletConfigKey, _ := getManagedKubeletConfigKey(mcp, f.client, kc1)
528:                    kubeletConfigKey, _ := getManagedKubeletConfigKey(mcp, f.client, kc1)
562:                    kubeletConfigKey, _ := getManagedKubeletConfigKey(mcp, f.client, kc1)
724:                    kubeletConfigKey, _ := getManagedKubeletConfigKey(mcp, f.client, kc1)
915:                    kubeletConfigKey, _ := getManagedKubeletConfigKey(mcp, f.client, kc1)

pkg/controller/node/node_controller_test.go
190:            e, _ := expected.(core.CreateAction)
195:            e, _ := expected.(core.UpdateAction)
200:            e, _ := expected.(core.PatchAction)

pkg/daemon/daemon_test.go
267:            e, _ := expected.(core.CreateAction)
276:            e, _ := expected.(core.UpdateAction)
285:            e, _ := expected.(core.PatchAction)

pkg/daemon/update.go
1630:   uid, _ := strconv.Atoi(osUser.Uid)
1640:   gid, _ := strconv.Atoi(osGroup.Gid)
```

reveals a lot of similar issues (though a few of those are valid), and some outside of tests.  So...agree this is worth fixing but I think let's get this in and do it as a followup?  I think there's some natural and normal tension around asking contributors to fixing existing present issues when working on the code.  Sometimes, it's really necessary (e.g. to add missing unit tests, as I did here).  But I also think it's better to avoid PRs growing too large and touching too many things; keep up momentum, reward contributors etc.